search/search-client-connect/src/main/java/io/camunda/search/connect/os/OpensearchConnector.java

```diff
@@ -125,14 +152,21 @@ private HttpHost getHttpHost(final ConnectConfiguration osConfig) {
     try {
       final var uri = new URI(osConfig.getUrl());
       return new HttpHost(uri.getScheme(), uri.getHost(), uri.getPort());
-    } catch (URISyntaxException e) {
+    } catch (final URISyntaxException e) {
       throw new SearchClientConnectException("Error in url: " + osConfig.getUrl(), e);
     }
   }
 
   protected HttpAsyncClientBuilder configureHttpClient(
-      final HttpAsyncClientBuilder httpAsyncClientBuilder, final ConnectConfiguration osConfig) {
+      final HttpAsyncClientBuilder httpAsyncClientBuilder,
+      final ConnectConfiguration osConfig,
+      final HttpRequestInterceptor... interceptors) {
```

Comment:
❓ why an array of interceptors here, when we only ever pass one?

Reply:
There was a case when configuration required multiple plugin repositories / interceptors, so we decided that we will support an array of interceptors. In addition to that, the plugin functionality will grow in similar way as Optimize C7 had.